src/herbivoresAndCarnivores.js

```diff
@@ -2,14 +2,45 @@
 
 class Animal {
   // write your code here
+  constructor(name) {
+    this.name = name;
+    this.health = 100;
+    Animal.alive.push(this);
+  }
 }
 
+Animal.alive = [];
+
 class Herbivore extends Animal {
   // write your code here
+  constructor(name) {
+    super(name);
+    this.hidden = false;
+    Animal.alive.push(this);
+  }
+
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
   // write your code here
+  constructor(name) {
+    super(name);
+    Animal.alive.push(this);
+  }
+
+  bite(animal) {
+    if ((animal instanceof Herbivore) && (animal.hidden === false)) {
```

Comment:
you van make it short
```suggestion
    if ((animal instanceof Herbivore) && !animal.hidden) {
```